map-data/src/main/java/org/triplea/map/reader/MapElementReader.java

```diff
@@ -0,0 +1,66 @@
+package org.triplea.map.reader;
+
+import java.io.InputStream;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import lombok.experimental.UtilityClass;
+import org.triplea.map.data.elements.GameTag;
+
+@UtilityClass
+public class MapElementReader {
+
+  public static MapReadResult readXml(final String xmlFile, final InputStream inputStream) {
+    try {
+      final GameTag gameTag = readXmlThrowing(inputStream);
+      return MapReadResult.builder().gameTag(gameTag).build();
+    } catch (final XMLStreamException e) {
+      return MapReadResult.builder()
+          .errorMessage(
+              "Invalid XML, parse error reading: "
+                  + xmlFile
+                  + "\n"
+                  + formatErrorMessage(new StringBuffer(), e))
+          .build();
+    }
+  }
+
+  private static GameTag readXmlThrowing(final InputStream inputStream) throws XMLStreamException {
+
+    final XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+    final XMLStreamReader streamReader = inputFactory.createXMLStreamReader(inputStream);
+
+    try {
+      return doRead(streamReader);
+    } finally {
+      streamReader.close();
+    }
+  }
+
+  private static GameTag doRead(final XMLStreamReader streamReader) throws XMLStreamException {
+
+    final int eventType = streamReader.next();
+
+    switch (eventType) {
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6102557)